### PR TITLE
fix(docs): render Mermaid diagrams with accessible titles and descriptions

### LINKED ISSUE #2677: Accessibility issue: Mermaid diagrams render as source code and expose arrow syntax to screen readers
## Summary

Mermaid diagrams on the HVE Core documentation site render as syntax-highlighted source code rather than diagrams with accessible titles and descriptions.

During manual NVDA testing of **Release Process → How Releases Work**, NVDA announced the Mermaid connection operator `-->` using punctuation terms such as "dash dash greater." It did not announce the content as a release workflow or communicate the connections between its stages.

This was discovered during the manual assistive-technology pass for #2566, but the affected page and Docusaurus Mermaid configuration are not part of that pull request's `origin/main...HEAD` diff.

## Steps to reproduce

1. Build and serve the Docusaurus documentation site.
2. Start NVDA and enable browse mode.
3. Open `/hve-core/docs/contributing/release-process`.
4. Navigate to **How Releases Work**.
5. Read the diagram content.
6. Repeat under **Maturity Lifecycle**.

## Observed behavior

- Each Mermaid fence renders as a `<pre><code class="language-mermaid">` block.
- Edge exposes the raw Mermaid program through the accessibility tree.
- NVDA reads operators such as `-->` as punctuation, including "greater."
- The diagrams have no computed accessible diagram title or description.
- The numbered steps following **How Releases Work** provide a detailed textual sequence, but users encounter the raw Mermaid source first.

The repository currently contains 69 Mermaid fence occurrences, so other documentation pages may have the same behavior.

## Expected behavior

Mermaid fences render as diagrams. Each meaningful diagram exposes a concise accessible title and description, and complex relationships have equivalent textual content. NVDA announces the diagram's purpose instead of Mermaid operators.

## Environment

- Windows 11
- Microsoft Edge 151.0.4129.72
- NVDA 2026.1.1
- NVDA browse mode
- Local Docusaurus production build

## Root cause

- `docs/docusaurus/docusaurus.config.js` does not enable Mermaid rendering or register the Docusaurus Mermaid theme.
- `docs/docusaurus/package.json` does not declare `@docusaurus/theme-mermaid`.

Do not replace `-->` with a Unicode arrow. The operator is valid Mermaid syntax; the defect is that the source is published rather than rendered.

## Proposed remediation

1. Add `@docusaurus/theme-mermaid` at the same version as the existing Docusaurus packages.
2. Enable Mermaid in the Docusaurus Markdown configuration and register the theme.
3. Add Mermaid `accTitle` and `accDescr` directives to both diagrams in `docs/contributing/release-process.md`.
4. Introduce each diagram with prose explaining its purpose.
5. Keep the numbered release steps as detailed equivalent content.
6. Audit the remaining Mermaid fences for accessible titles, descriptions, and equivalent content.
7. Add production-build regression coverage that detects intended Mermaid diagrams rendered as raw `language-mermaid` code.

## Acceptance criteria

- [ ] Mermaid rendering is enabled in the Docusaurus site.
- [ ] `@docusaurus/theme-mermaid` matches the version of the other Docusaurus packages.
- [ ] Intended diagrams do not appear as raw `pre code.language-mermaid` content in the production build.
- [ ] **How Releases Work** has an accessible title and description.
- [ ] **Maturity Lifecycle** has an accessible title and description.
- [ ] The numbered release steps remain available as detailed equivalent content.
- [ ] NVDA no longer reads `-->` as "dash dash greater" when navigating these diagrams.
- [ ] NVDA announces meaningful diagram titles and descriptions.
- [ ] Diagrams remain legible in light, dark, forced-colors, and supported zoom states.
- [ ] Automated coverage verifies that Mermaid rendering is active.
- [ ] A broader audit identifies Mermaid diagrams without equivalent textual content.
- [ ] Docusaurus build, unit tests, accessibility checks, and Playwright tests pass.
- [ ] A qualified accessibility reviewer confirms the result before closure.

## Accessibility assessment

- Manual observation: raw Mermaid source and arrow operators are announced instead of a named diagram.
- Candidate criteria: WCAG 2.2 SC 1.1.1, Non-text Content, and SC 1.3.1, Information and Relationships.

The criterion mapping and final conformance result remain subject to qualified accessibility review.

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
# Pull Request

## Description

Mermaid fences on the documentation site shipped as syntax-highlighted source blocks rather than diagrams. Screen readers announced the raw Mermaid program — NVDA read the `-->` connection operator as "dash dash greater" — and the graphics carried no accessible name or description. This enables Mermaid rendering, gives every deployed diagram an authored title and description, and adds validation that keeps the corpus complete.

> Automation here can prove that a title and description exist, render, and associate correctly. It cannot prove that they *mean* the right thing. The semantic pass is called out under Testing and remains open.

### Rendering configuration and dependencies

Enabling the renderer was the prerequisite for everything else, since accessibility directives are inert while fences ship as code blocks.

- Set `markdown.mermaid` and registered **@docusaurus/theme-mermaid** as the first entry in `themes` in *docusaurus.config.js*, with a `themeConfig.mermaid` block binding the `neutral` and `dark` palettes to the site's existing color modes.
- Added two exact-pinned runtime dependencies: `@docusaurus/theme-mermaid` at *3.10.2* and `mermaid` at *11.16.1*, matching the file's existing no-range convention.
- Regenerated *package-lock.json* with roughly 110 net-new transitive entries — the d3 family, cytoscape and its layout plugins, dagre-d3-es, roughjs, katex, marked, dompurify, and `@mermaid-js/parser`. Every entry resolves from the canonical public npm registry with `integrity` and `license` recorded.

### Diagram metadata across the documentation corpus

- Authored `accTitle` and `accDescr` pairs across **44 documentation files**, covering agent guides, architecture pages, ADRs, BRDs, PRDs, security documentation, and templates.
- The corpus now holds **64 fences**, each with exactly one active title and description. Families in use are `flowchart`, legacy `graph`, `stateDiagram-v2`, `journey`, and `erDiagram`.
- Repaired four `stateDiagram-v2` transition labels in *agentic-workflows.md* that used a semicolon. A semicolon terminates a Mermaid statement, so those labels failed to parse once rendering was live. The pre-repair form is now the parse-invalid negative control in the e2e spec.
- Refreshed `ms.date` on every edited file.

### Validation

Three layers were added so that a diagram authored later cannot silently ship without metadata.

- *validate-mermaid-accessibility.mjs* discovers deployed pages through `git ls-files`, filters to the Docusaurus deployment boundary, extracts fences with matched markers, and requires exactly one active `accTitle` and `accDescr` per fence. Directives inside `%%` comments do not count. Failures report as `file:line: message` with a non-zero exit; `--json` emits the inventory for downstream consumers.
- *validate-mermaid-accessibility.test.mjs* covers the pure helpers with eight `node:test` cases, including the deployment-boundary predicate, unclosed-fence errors, multi-line `accDescr` flattening, and rejection of a commented-out title.
- *mermaid-accessibility.spec.ts* consumes that same `--json` inventory, so no second hand-maintained diagram list exists. It renders every fence in Chrome and asserts each SVG exposes `role="graphics-document"` with `aria-labelledby` and `aria-describedby` resolving to non-empty `title` and `desc` whose text matches the authored source. Two negative controls guard against passing by vacuity: a parse-invalid fixture the browser must reject, and a broken ARIA association the assertion helper must reject. Four representative routes then assert exact diagram counts, zero remaining `pre code.language-mermaid` blocks, and resolvable accessible names on the published pages.
- Wired `test:mermaid-a11y` into `test` and `test:coverage`, and excluded the native ESM validator test from Jest discovery so `node --test`, Jest, and Playwright each own a distinct layer.

## Related Issue(s)

Fixes #2677

## Type of Change

Select all that apply:

**Code & Documentation:**

* [x] Bug fix (non-breaking change fixing an issue)
* [ ] New feature (non-breaking change adding functionality)
* [ ] Breaking change (fix or feature causing existing functionality to change)
* [x] Documentation update

**Infrastructure & Configuration:**

* [ ] GitHub Actions workflow
* [ ] Linting configuration (markdown, PowerShell, etc.)
* [ ] Security configuration
* [ ] DevContainer configuration
* [x] Dependency update

**AI Artifacts:**

* [ ] Reviewed contribution with `hve-builder` and addressed all actionable findings
* [ ] Copilot instructions (`.github/instructions/*.instructions.md`)
* [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
* [ ] Copilot agent (`.github/agents/*.agent.md`)
* [ ] Copilot skill (`.github/skills/*/SKILL.md`)
* [ ] Copilot hook (`.github/hooks/*/*.json`)
* [ ] Eval spec added/updated for changed AI artifacts (`evals/`)

> Note for AI Artifact Contributors:
>
> * Agents: Research, indexing/referencing other project (using standard VS Code GitHub Copilot/MCP tools), planning, and general implementation agents likely already exist. Review `.github/agents/` before creating new ones.
> * Skills: Must include both bash and PowerShell scripts. See [Skills](../docs/contributing/skills.md).
> * Model Versions: Contributions **MUST** target models listed in the model catalog (`scripts/linting/model-catalog.json`) whose provider appears in `providerAllowlist` and whose status is `ga` or `preview`. Run `npm run lint:models` to validate references.
> * See [Agents Not Accepted](../docs/contributing/custom-agents.md#agents-not-accepted) and [Model Version Requirements](../docs/contributing/ai-artifacts-common.md#model-version-requirements).

**Other:**

* [ ] Script/automation (`.ps1`, `.sh`, `.py`)
* [ ] Other (please describe):

## Sample Prompts (for AI Artifact Contributions)

**User Request:**

**Execution Flow:**

**Output Artifacts:**

**Success Indicators:**

For detailed contribution requirements, see:

* Common Standards: [docs/contributing/ai-artifacts-common.md](../docs/contributing/ai-artifacts-common.md) - Shared standards for XML blocks, markdown quality, RFC 2119, validation, and testing
* Agents: [docs/contributing/custom-agents.md](../docs/contributing/custom-agents.md) - Agent configurations with tools and behavior patterns
* Prompts: [docs/contributing/prompts.md](../docs/contributing/prompts.md) - Workflow-specific guidance with template variables
* Instructions: [docs/contributing/instructions.md](../docs/contributing/instructions.md) - Technology-specific standards with glob patterns
* Skills: [docs/contributing/skills.md](../docs/contributing/skills.md) - Task execution utilities with cross-platform scripts

## Testing

Automated validation run on this branch:

| Check | Result |
|-------|--------|
| `npm ci` (docs/docusaurus) | Passed — 2053 packages, manifest and lockfile in sync |
| `npm run test:mermaid-a11y` | Passed — 8 native tests, 64 fences validated |
| `npm run typecheck` (docs/docusaurus) | Passed |
| `npm run lint:a11y` (docs/docusaurus) | Passed |
| `npm run validate:local` | 26 of 27 checks passed; `lint:md-links` failed on three unrelated skill reference files |
| `npm run validate:docs` | Passed — docs lint, label registry, typecheck, 101 Jest tests, and the Mermaid source lane |
| `npm run spell-check` | Passed — 817 files, 0 issues |
| `npm run lint:md-links` | Skipped by explicit direction |

The `lint:md-links` failures are in `.github/skills/project-planning/security-planning/references/data-classification.md`, `.github/skills/rai/rai-standards/references/eu-ai-act.md`, and `.github/skills/rai/rai-standards/SKILL.md`. None of those files is in this change set. Because `validate:local` is a sequential chain, that failure initially masked the 19 checks that follow it; those were run individually and all passed.

Diff-based assessment confirmed the four asserted route counts against source (Customization 1, Release Process 1, Agentic Workflows 3, SSSC Planner PRD 5) and corrected two figures that had gone stale against current `main`: the inventory lock moved from 65 to 64, and Release Process from 2 diagrams to 1, after `main` removed the Maturity Lifecycle state diagram. An Agentic Workflows route was added so `stateDiagram` stays covered at the route tier.

The full Playwright documentation lane is CI-owned and was not run locally on this branch.

> [!IMPORTANT]
> Manual assistive-technology verification has not been performed on this branch. The automated layers prove that every diagram renders as a named, described graphic with correct ARIA associations; they cannot prove that a title conveys the right purpose or that a description is equivalent to the diagram it replaces. A qualified accessibility reviewer should confirm the 64 title and description pairs, the complex-diagram prose equivalents, and representative screen-reader, forced-colors, and zoom behavior before this is considered conformant.

### Required Checks

* [x] Documentation is updated (if applicable)
* [x] Files follow existing naming conventions
* [x] Changes are backwards compatible (if applicable)
* [x] Tests added for new functionality (if applicable)

### AI Artifact Contributions

* [ ] Used `hve-builder` review mode to review contribution
* [ ] Addressed all actionable findings from the `hve-builder` review
* [ ] Verified contribution follows common standards and type-specific requirements

### Required Local Checks

The following local-safe validation commands must pass before merging:

* [ ] Local validation aggregate: `npm run validate:local` (26 of 27 checks passed; `lint:md-links` failed on three files outside this change set)
* [x] Documentation validation (if docs changed): `npm run validate:docs`
* [x] Spell checking: `npm run spell-check`
* [ ] Link validation: `npm run lint:md-links` (skipped by explicit direction)

## Security Considerations

* [x] This PR does not contain any sensitive or NDA information
* [x] Any new dependencies have been reviewed for security issues
* [ ] Security-related scripts follow the principle of least privilege (N/A — no files under `scripts/security/` changed)

Both new scripts call `execFileSync` in argument-array form with no shell, so no command-injection surface exists, and the e2e spec invokes `process.execPath` rather than a `PATH`-resolved binary. File paths originate from `git ls-files` scoped to `docs`, are normalized, allowlist-filtered, and joined against a resolved repository root before any read. Neither script writes files, opens network connections, or reads credentials. Mermaid initializes with `securityLevel: 'strict'`, and the browser bundle loads from local `node_modules` rather than a CDN. Both added dependencies are exact-pinned and resolve from the canonical public npm registry.

## Additional Notes

The inventory lock and per-route diagram counts are deliberate but create a maintenance coupling: authoring or removing a Mermaid fence anywhere under `docs/` requires updating `expect(inventory).toHaveLength(64)`, and editing one of the four asserted pages requires updating its `diagramCount`. The expected family list is a closed set, so introducing a `sequenceDiagram` or `gantt` fence requires editing that assertion as well. The failure message names the offending file and line, so the required edit is discoverable from the test output.

One rough edge worth a follow-up: the validator's `--json` mode emits no JSON when failures exist, so a failing corpus surfaces in the e2e spec as a module-load exception during collection rather than a readable test failure.

### CODE REVIEW

**Review by jkim323 — APPROVED** [review]

File: docs/docusaurus/e2e/mermaid-accessibility.spec.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+const packageRoot = path.resolve(__dirname, '..');
+const inventory = JSON.parse(execFileSync(
+  process.execPath,
+  ['scripts/validate-mermaid-accessibility.mjs', '--json'],
+  { cwd: packageRoot, encoding: 'utf8' },
+)) as MermaidFence[];
+const mermaidBrowserBundle = path.join(packageRoot, 'node_modules/mermaid/dist/mermaid.min.js');
+
+const routeCases = [
```
Comment (katriendg) [inline docs/docusaurus/e2e/mermaid-accessibility.spec.ts]: The exhaustive loop renders source through Mermaid directly, while the production Docusaurus path covers only these four routes (10 of 64 diagrams). Consider deriving route cases from the inventory so every deployed fence is also checked for rendered SVG output and zero remaining `pre code.language-mermaid` blocks.

File: docs/agents/rai-planning/handoff-pipeline.md [hunk 2]
```diff
@@ -66,6 +68,8 @@ Phase 6 produces a review summary covering observations across six dimensions. T
 
 ```mermaid
 flowchart TD
+  accTitle: RAI Review Summary Synthesis
+  accDescr: Six RAI planning phases contribute standards alignment, threat completeness, control effectiveness, evidence quality, risk classification, and tradeoff resolution to the final review.
```
Comment (katriendg) [inline docs/agents/rai-planning/handoff-pipeline.md]: This description says six planning phases contribute, but the diagram shows Phases 1 through 5 feeding six review dimensions. Consider describing those relationships directly so the text alternative matches the visual process model.

File: docs/architecture/agentic-workflows.md [hunk 3]
```diff
@@ -23,6 +23,8 @@ hve-core uses GitHub Agentic Workflows to support the journey from issue creatio
 
 ```mermaid
 flowchart TD
+    accTitle: Issue Triage and Pull Request Review Workflow
+    accDescr: Six connected workflows move an issue through triage, implementation, automated review outcomes, revision loops, human approval, and merge.
```
Comment (katriendg) [inline docs/architecture/agentic-workflows.md]: This diagram depicts issue triage, implementation, PR review, and human review rather than six connected workflows. Consider naming only the workflows and outcomes actually shown so the accessible description stays equivalent to the graphic.

File: docs/customization/README.md [hunk 4]
```diff
@@ -38,6 +38,8 @@ HVE Core supports a range of customization depths. Start with the lightest optio
 
 ```mermaid
 graph LR
+  accTitle: HVE Core Customization Depth Spectrum
+  accDescr: Customization progresses from VS Code settings through instructions, agents and prompts, skills, marketplace recipes, build-system changes, and a full fork as complexity increases.
```
Comment (katriendg) [inline docs/customization/README.md]: The diagram and adjacent table use `Plugin Manifest` at this stage, while the description says `marketplace recipes`. Consider using `the plugin manifest` here so visual and non-visual readers receive the same customization path.

File: docs/planning/prds/sssc-planner.md [hunk 5]
```diff
@@ -206,6 +212,8 @@ flowchart TD
 
 ```mermaid
 erDiagram
+  accTitle: SSSC Session State and Artifact Model
+  accDescr: An SSSC session persists state, produces phase artifacts, processes references, records notices, generates handoff items, and optionally traces back to PRD state.
```
Comment (katriendg) [inline docs/planning/prds/sssc-planner.md]: The diagram shows `PRD` tracking `PRD_STATE` as a separate relationship, with no edge connecting either entity to `SSSC_SESSION`. Consider adding the intended traceability edge or removing the traceability claim from this description.

**Review by katriendg — APPROVED** [review]
The Mermaid renderer, source validation, and SVG metadata associations are implemented well, and the focused validator passes all 64 fences. The review raised five findings, mostly medium and some may be considered optional. Leaving inline comments where possible.
Approving as I know you tackle each one. Thanks.

File: docs/planning/prds/sssc-planner.md (hunk 5 above)
Comment (WilliamBerryiii) [reply docs/planning/prds/sssc-planner.md]: Addressed in `0e011d18`. I added an optional `PRD_STATE` to `SSSC_SESSION` seed relationship and updated the description to state that PRD state can optionally seed a session, while the session persists its own state and artifacts.

File: docs/docusaurus/e2e/mermaid-accessibility.spec.ts (hunk 1 above)
Comment (WilliamBerryiii) [reply docs/docusaurus/e2e/mermaid-accessibility.spec.ts]: Addressed in `0e011d18`. The production route cases are now grouped directly from the validator JSON inventory, including README index routing, so all 47 Mermaid-bearing pages and all 64 diagrams are covered without a second manual list. The test continues to assert accessible SVG metadata and zero remaining raw Mermaid source blocks.

File: docs/agents/rai-planning/handoff-pipeline.md (hunk 2 above)
Comment (WilliamBerryiii) [reply docs/agents/rai-planning/handoff-pipeline.md]: Addressed in `0e011d18`. The description now states that Phases 1 through 5 feed the six named review dimensions and that all six dimensions feed the final review summary.

File: docs/customization/README.md (hunk 4 above)
Comment (WilliamBerryiii) [reply docs/customization/README.md]: Addressed in `0e011d18`. The accessible description now uses `the plugin manifest`, matching the diagram and adjacent table.

File: docs/architecture/agentic-workflows.md (hunk 3 above)
Comment (WilliamBerryiii) [reply docs/architecture/agentic-workflows.md]: Addressed in `0e011d18`. The description now follows the relationships shown: issue triage routes qualifying issues into implementation, then pull requests move through automated review outcomes, revision loops when needed, human review, and merge.

**Review by jkim323 — APPROVED** [review]

**Review by agreaves-ms — APPROVED** [review]
